Honor SERVO_ENABLE_DEBUG_ASSERTIONS on the build machines.

emilio opened the pull request:
<!-- Please describe your changes on the following line: -->

As part of #13127.

cc @aneeshusa 

---

<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/13387)

<!-- Reviewable:end -->

emilio committed:
build: Honor SERVO_ENABLE_DEBUG_ASSERTIONS.

highfive commented:
Heads up! This PR modifies the following files:
 * @wafflespeanut: python/servo/build_commands.py

emilio commented:
@bors-servo: try

Not sure if the new build config has been rolled out already.

bors-servo commented:
:hourglass: Trying commit 16771bc with merge de58d53...

bors-servo commented:
:broken_heart: Test failed - [arm64](http://build.servo.org/builders/arm64/builds/3078)

Ms2ger commented:
Do we really want an env var for this? Arguments to mach are easier to document and discover.

emilio commented:
@bors-servo: try

bors-servo commented:
:hourglass: Trying commit 8266e40 with merge 4aa4372...

emilio commented:
@Ms2ger: You're right, see the `TODO`.

The reason why this is done this way is to prevent shipping a change in saltfs that could potentially cause a lot of perma-failures. We added the env var first, which is a no-op, then see the result of running the tests with debug assertions in another servo PR (this one).

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3311)

emilio commented:
@bors-servo: retry (build was interrupted)

bors-servo commented:
:hourglass: Trying commit 8266e40 with merge 7fab83e...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3313)

highfive commented:
▶ Unexpected subtest result in /_mozilla/mozilla/promise.html:
      │ FAIL [expected PASS] Native promise from async callback can be resolved
      │   → assert_true: expected true got false
      │ 
      └ @http://web-platform.test:8000/_mozilla/mozilla/promise.html:58:11

emilio commented:
@bors-servo: try

bors-servo commented:
:hourglass: Trying commit 537af57 with merge f147e4a...

bors-servo commented:
:snowman: The build was interrupted to prioritize another pull request.

bors-servo commented:
:hourglass: Trying commit 537af57 with merge e14d5a1...

bors-servo commented:
:sunny: Test successful - [arm32](http://build.servo.org/builders/arm32/builds/3074), [arm64](http://build.servo.org/builders/arm64/builds/3091), [linux-dev](http://build.servo.org/builders/linux-dev/builds/3145), [linux-rel](http://build.servo.org/builders/linux-rel/builds/3319), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/3145), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3141), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3268), [windows-dev](http://build.servo.org/builders/windows-dev/builds/1037)

aneeshusa commented:
@larsbergstrom is rolling out the env var right now.

larsbergstrom commented:
Buildbot is scheduled to restart at the next quiescent point, at which time this should be rolled out.

aneeshusa commented:
@bors-servo retry
- The queue is now empty, so Buildbot should have restarted.

bors-servo commented:
:hourglass: Trying commit 537af57 with merge 78bad5e...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3341)

aneeshusa commented:
Looks like this is working now, so r=me after removing the second commit with the testing `debug_assert`.

emilio commented:
@bors-servo: r=aneeshusa

Let's see how many timeouts.

bors-servo commented:
:pushpin: Commit 1de6a8b has been approved by `aneeshusa`

<!-- @bors-servo r=aneeshusa 1de6a8be67f4f13f1c91b184165047dba35d2159 -->

bors-servo commented:
:hourglass: Testing commit 1de6a8b with merge 890ab57...

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3347)

emilio committed:
script: Fix bogus is_listed_element assertion.

emilio committed:
layout: Use the fragment's writing mode when computing sizes relevant to that fragment.

Otherwise we might mix writing modes. Not totally sure this change is correct in
the case we're mixing them, we might need to just not checking that operation.

emilio commented:
@bors-servo: try

bors-servo commented:
:hourglass: Trying commit a84b0df with merge 0c40f45...

emilio commented:
@bors-servo: try

bors-servo commented:
:hourglass: Trying commit a929fcb with merge c91bb2b...

emilio commented:
@nox or @jdm or @Ms2ger: r? On the second commit.
@pcwalton: r? On the last one.

bors-servo commented:
:broken_heart: Test failed - [linux-rel](http://build.servo.org/builders/linux-rel/builds/3384)

emilio committed:
script: Prevent an integer overflow that was hitting us on htmlimageelement.rs

emilio commented:
@bors-servo: try

bors-servo commented:
:hourglass: Trying commit 5d0c804 with merge 512d5d5...

bors-servo commented:
:broken_heart: Test failed - [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3327)

emilio commented:
@bors-servo: retry

bors-servo commented:
:hourglass: Trying commit 5d0c804 with merge 47ed107...

bors-servo commented:
:sunny: Test successful - [arm32](http://build.servo.org/builders/arm32/builds/3155), [arm64](http://build.servo.org/builders/arm64/builds/3169), [linux-dev](http://build.servo.org/builders/linux-dev/builds/3239), [linux-rel](http://build.servo.org/builders/linux-rel/builds/3416), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/3224), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3210), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3349), [windows-dev](http://build.servo.org/builders/windows-dev/builds/1120)

emilio commented:
@bors-servo: retry
 
 * Try with the new renderer, just in case.

bors-servo commented:
:hourglass: Trying commit 5d0c804 with merge 55334f7...

bors-servo commented:
:boom: Test timed out

jdm commented:
@bors-servo: retry

bors-servo commented:
:hourglass: Trying commit 5d0c804 with merge 4895d91...

bors-servo commented:
:sunny: Test successful - [arm32](http://build.servo.org/builders/arm32/builds/3198), [arm64](http://build.servo.org/builders/arm64/builds/3211), [linux-dev](http://build.servo.org/builders/linux-dev/builds/3283), [linux-rel](http://build.servo.org/builders/linux-rel/builds/3462), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/3275), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3256), [mac-rel-wpt](http://build.servo.org/builders/mac-rel-wpt/builds/3399), [windows-dev](http://build.servo.org/builders/windows-dev/builds/1161)

emilio commented:
review ping @jdm/@Ms2ger for the image setter bits and and @pcwalton/@mbrubeck for the writing mode changes?

jdm commented on components/script/dom/htmlimageelement.rs:
```diff
@@ -435,14 +435,24 @@ impl VirtualMethods for HTMLImageElement {
 
 fn image_dimension_setter(element: &Element, attr: Atom, value: u32) {
+    use app_units::AU_PER_PX;
+    use std::i32;
```
Can we move these to the top of the file?

emilio replied:
Done

jdm commented on components/script/dom/htmlimageelement.rs:
```diff
@@ -435,14 +435,24 @@ impl VirtualMethods for HTMLImageElement {
     };
-    let dim = LengthOrPercentageOrAuto::Length(Au::from_px(value as i32));
+
+    let pixel_value = if value > (i32::MAX / AU_PER_PX) as u32 {
```
I wonder if this belongs in Au::from_px, or whether Au::from_px should return an Option value instead? I don't think that I'm a good person to make that call.

emilio replied:
Yeah, I'll open an issue for this in app_units, and mention it from here.

I'd still like to land this before that is resolved, since that's potentially a big change, and I think the benefits of this are worth it.

emilio replied:
Filled https://github.com/servo/app_units/issues/22

emilio commented:
@bors-servo: r=aneeshusa,jdm,pcwalton

```
20:43 <•pcwalton> emilio: looks good
20:43 <•pcwalton> r=me if you need it
20:43 <emilio> jdm: do the changes to the image setter bits look ok to you?
20:44 <•jdm> emilio: link me again so I can refresh my memory?
20:44 <emilio> jdm: https://github.com/servo/servo/pull/13387/commits/fdd84713105b5c87f7adde1319711966d00e5b66
20:45 <•jdm> emilio: yes
```

Finally!

bors-servo commented:
:pushpin: Commit fdd8471 has been approved by `aneeshusa,jdm,pcwalton`

<!-- @bors-servo r=aneeshusa,jdm,pcwalton fdd84713105b5c87f7adde1319711966d00e5b66 -->

emilio commented:
(Thanks for the reviews btw!)

bors-servo commented:
:hourglass: Testing commit fdd8471 with merge 19a5a30...

bors-servo commented:
:sunny: Test successful - [arm32](http://build.servo.org/builders/arm32/builds/3316), [arm64](http://build.servo.org/builders/arm64/builds/3329), [linux-dev](http://build.servo.org/builders/linux-dev/builds/3406), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/23), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/23), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/3393), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/3373), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/74), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/73), [windows-dev](http://build.servo.org/builders/windows-dev/builds/1281)
